### ISSUE TITLE
The service header cannot accommodate longer titles

### ISSUE THREAD

**Issue by bartfeenstra** [nhsuk/nhsuk-frontend#708]
# Bug Report

## What is the issue?

When using a longer title such as *Hampshire & Isle of Wight Saliva Test*, the service header does not wrap lines for narrower pages, causing the title to collide with the menu and search buttons.

## What steps are required to reproduce the issue?

Use the service header and use a longer site title such as *Hampshire & Isle of Wight Saliva Test*. Slowly make the page narrower and observe the interaction between the title, and the menu and search buttons. You'll observe the buttons going below the title instead of staying in the upper right-hand corner and the title wrapping onto multiple lines.

## What was the environment where this issue occurred?

* NHS.UK frontend package version: 4.1.0
* Browsers: Firefox 85.0.1 (Linux), and Chromium 88.0.4324.182 (Linux)

My CSS skills are quite mediocre so I won't be able to provide a good solution myself. However, please let me know if I should provide additional details or when I should test any solutions.

**Comment by AdamChrimes** [comment]
Hey @bartfeenstra thanks for opening this issue, the team will have a look at it and get back to you once we've had a look and a bit of tinker with the code!

**Comment by AdamChrimes** [comment]
This seems to be an issue between tablet (641px) and large-desktop (990px) - dependant on the length of the service name.

<img width="673" alt="Screenshot 2021-02-22 at 16 02 10" src="https://user-images.githubusercontent.com/11615501/108734382-537ff080-7527-11eb-96d7-bef9c11a9c9c.png">

(Screenshot at 641px)

In this case, with "Hampshire & Isle Wight" being the name of the NHS organisation and "Salvia test" being the name of the service, we could use the [Organisational header](https://service-manual.nhs.uk/design-system/components/header#organisational-header) and extend it to include a service name. Is this something that would work for you in this case @bartfeenstra? We will still need to look into the longer service name bug regardless but this might be of use for now.

<img width="673" alt="Screenshot 2021-02-22 at 16 07 41" src="https://user-images.githubusercontent.com/11615501/108735157-18ca8800-7528-11eb-9702-e5a97b6aa3c7.png">

(Screenshot at 641px)

Nunjucks markup

```

{{ header({
    "showNav": "true",
    "showSearch": "true",
    "organisation": {
      "name": "Hampshire &",
      "split": "Isle of Wight Saliva Test",
      "descriptor": "NHS Foundation Trust"
    },
    "primaryLinks": [
      {
        "url"  : "#",
        "label" : "Your hospital visit"
      },
      {
        'url' : '#',
        'label' : 'Wards and departments'
      },
      {
        'url'  : '#',
        'label' : 'Conditions and treatments'
      },
      {
        'url'  : '#',
        'label' : 'Our people'
      },
      {
        'url' : '#',
        'label' : 'Our research'
      }
    ]
  })
}}
```

**Comment by bartfeenstra** [comment]
I came back to report that this happens when there's a search bar, which I'm (ab)using for a tiny user navigation menu (log in/out) to see your response.

There's technically no organization as such behind this site. It's a joint project between a Trust, a university, and a council. That doesn't mean I can't just shove it in the organizational header regardless, but the "Saliva Test" is a key part of the service name. How easy would it be to make the title and menu/search share the space, and have the title wrap?

**Comment by AdamChrimes** [comment]
Adding a `max-width` to the `nhsuk-header__service-name` below the `large-desktop` breakpoint seems to do the trick

<img width="697" alt="Screenshot 2021-02-22 at 17 35 29" src="https://user-images.githubusercontent.com/11615501/108746540-5cc38a00-7534-11eb-8af8-7fba1ebca3c8.png">

Sass

```
.nhsuk-header__service-name {

  @include mq($until: large-desktop) {
    max-width: 220px;
  }

}
```

I'll have to run this past our designer to check whether the width is the correct value, but this is something you could try adding as an extra style to your individual app to test for us.

**Comment by AdamChrimes** [comment]
This will be fixed in release 5.0.0 (next week)

**Comment by bartfeenstra** [comment]
Thanks! In the meantime we stopped abusing the search form for additional navigation (log in/out) so I hadn't checked back here for a while. Thanks for picking this up, though! :)
